server/app.js

```diff
@@ -0,0 +1,40 @@
+const express = require('express');
+const path = require('path');
+const bodyParser = require('body-parser');
+const cookieParser = require('cookie-parser');
+// const router = require('./controllers');
+const { authCheck } = require('./authentication/authentication');
+
+const app = express();
+app.set('port', process.env.PORT || 4000);
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(express.static(path.join(__dirname, '..', 'public')));
+app.use(cookieParser());
+
+app.use((req, res, next) => {
+  authCheck(req, (authErr, token) => {
```

Comment:
Once you have some app routes, you might need to refactor this authCheck to be a per-router middleware, instead of an application-level middleware, because most routes will use authCheck but not all of them - for example, POSTing to signup.

Here are some links which explain a little more:
https://expressjs.com/en/guide/using-middleware.html#middleware.router
https://scotch.io/tutorials/route-middleware-to-check-if-a-user-is-authenticated-in-node-js

Reply:
will do,